### PR TITLE
[Add] signup now fetch to server

### PR DESCRIPTION
**Pull request by wood-crafter**

### CODE REVIEW

File: src/providers/useAuth.js [hunk 1]
```diff
@@ -27,6 +27,19 @@ export const useAuth = () => {
         throw new Error('Credentials mismatch...!')
       }
     },
+    async attemptSignup(username, password) {
+      const {status_code} = await postData('https://alpha-sneu.xyz/api/v1/users/signup', {username, password})
+
+      if(status_code === 200) {
```
Comment (hoangph271) [inline src/providers/useAuth.js]: Maybe rewrite this part...?

```js
// This should be imported from another module
const API_ROOT = 'https://alpha-sneu.xyz/api/v1'

// Rename this `status_code` into JS style
const { status_code: statusCode } = await postData(`${API_ROOT}/users/signup`, { username, password })
```

File: src/providers/useAuth.js [hunk 2]
```diff
@@ -27,6 +27,19 @@ export const useAuth = () => {
...
+        setUser({username})
+      }
+      if(status_code === 409) {
+        throw new Error('Signup infomation existed')
+      }
+      else {
+        throw new Error('Error')
```
Comment (hoangph271) [inline src/providers/useAuth.js]: In the response JSON there is a field called `errors`, use this as the error message...?

**Review by hoangph271 — CHANGES_REQUESTED** [review]
Check these comments out

File: src/providers/useAuth.js (hunk 2 above)
Comment (wood-crafter) [reply src/providers/useAuth.js]: I resolved

File: src/providers/useAuth.js (hunk 1 above)
Comment (wood-crafter) [reply src/providers/useAuth.js]: i resolved

**Pull request merged**